backend/sirius-components-diagrams-layout/src/main/java/org/eclipse/sirius/components/diagrams/layout/incremental/IncrementalLayoutEngine.java

```diff
@@ -73,12 +75,36 @@
 
     private final NodeSizeProvider nodeSizeProvider;
 
+    /**
+     * Class representing border node on a parent side.
+     */
+    private class BorderNodesOnSide {
+
+        private final List<NodeLayoutData> borderNodes;
+
+        private final RectangleSide side;
+
+        BorderNodesOnSide(RectangleSide side, List<NodeLayoutData> borderNodes) {
+            this.side = Objects.requireNonNull(side);
+            this.borderNodes = Objects.requireNonNull(borderNodes);
+        }
+
+        public List<NodeLayoutData> getBorderNodes() {
+            return this.borderNodes;
+        }
+
+        public RectangleSide getSide() {
+            return this.side;
+        }
+    }
```

Comment:
Stop with the creation of internal classes.

Reply:
done